pkg/controller/jobframework/jobframework.go

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package jobframework
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+
+	"github.com/go-logr/logr"
+	"k8s.io/apimachinery/pkg/api/meta"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	"sigs.k8s.io/kueue/pkg/controller/jobs/noop"
+	"sigs.k8s.io/kueue/pkg/util/cert"
+)
+
+var (
+	errFailedMappingResource = errors.New("restMapper failed mapping resource")
+)
+
+func SetupControllers(mgr ctrl.Manager, log logr.Logger, certsReady chan struct{}, opts ...Option) error {
+	// The controllers won't work until the webhooks are operating, and the webhook won't work until the
+	// certs are all in place.
+	cert.WaitForCertsReady(log, certsReady)
```

Comment:
Did you forget to remove this from main?

Or maybe it shouldn't be here.

Reply:
Both here and the main one is intended although I should replace https://github.com/kubernetes-sigs/kueue/blob/78acd5305dc023002f87a221afc32d1cd5ca32c3/cmd/kueue/main.go#L222-L226 with `cert.WaitForCertsReady().`

I think we need to perform `cert.WaitForCertsReady()` here since in the in-house custom job kueue-manager, only this function launches controllers and webhooks.

However, the kueue-manager launches controllers and webooks in this function and main's `setupControllers()`.

So, I think both Is needed. @alculquicondor WDYT?

Reply:
yeah, but maybe they didn't setup certs at all? I think they can call the function if they need it.

Reply:
It makes sense. Let's remove it from here. 
They can call `WaitForCertsReady()` since it is an exported one.

Reply:
Done.